src/validation.cpp

```diff
@@ -3888,7 +3888,9 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, const Co
         int commitpos = GetWitnessCommitmentIndex(block);
         if (commitpos != -1) {
             uint256 hashWitness = BlockWitnessMerkleRoot(block);
-            if ((block.vtx[0]->wit.vtxinwit.size() > 0 &&block.vtx[0]->wit.vtxinwit[0].scriptWitness.stack.size() != 1) || block.vtx[0]->wit.vtxinwit[0].scriptWitness.stack[0].size() != 32) {
```

Comment:
if ``block.vtx[0]->wit.vtxinwit.empty()``, the last term in this condition will cause an out-of-bound access to that array.